runners/mlcube_singularity/mlcube_singularity/singularity_run.py

```diff
@@ -1,85 +1,220 @@
+        Args:
+            mlcube: Current MLCube configuration. Contains all fields from MLCube configuration file (YAML) including
+                platform-specific configuration sections (docker/singularity). In addition, this dictionary contains
+                'runtime' configuration (`root` and `workspace`), and `runner` configuration that contains default
+                runner configuration (Singularity in this case) from system settings file (it is exact or modified
+                version of `Config.DEFAULT` dictionary to account for user local environment).
+        Idea is that if mlcube contains `singularity` section, then it means that we use it as is. Else, we can try
+        to run this MLCube using information from `docker` section if it exists.
+        """
+        s_cfg: t.Optional[DictConfig] = mlcube.get("singularity", None)
+        if not s_cfg:
+            # Singularity runner will try to use docker section. At this point, it will work as long as we assume we
+            # pull docker images from a docker hub.
+            logger.warning(
+                "SingularityRun singularity configuration not found in MLCube file (singularity=%s).",
+                str(s_cfg),
+            )
+
+            d_cfg = mlcube.get("docker", None)
+            if not d_cfg:
+                logger.warning(
+                    "SingularityRun docker configuration not found too. Singularity runner will likely "
+                    "fail to run."
+                )
+                return
+
+            # The idea is that we can use the remote docker image as a source for the build process, automatically
+            # generating an image name in a local environment. Key here is that the source has a scheme - `docker://`
+            # The --fakeroot switch is useful and is supported in singularity version >= 3.5
+            build_args = ""
+            try:
+                SingularityRun.check_install()
+                version: semver.VersionInfo = get_singularity_version_info()
+                logger.info("SingularityRun singularity version %s", str(version))
+                if version >= semver.VersionInfo(major=3, minor=5):
+                    logger.info(
+                        "SingularityRun will use --fakeroot CLI switch (version >= 3.5)."
+                    )
+                    build_args = "--fakeroot"
+                else:
+                    logger.warning(
+                        "SingularityRun will not use --fakeroot CLI switch (version < 3.5)"
+                    )
+            except:
+                logger.warning(
+                    "SingularityRun can't get singularity version (do you have singularity installed?)."
+                )
+
+            build_file = "docker://" + d_cfg["image"]
+            if "tar_file" in d_cfg:
+                build_file = "docker-archive:" + d_cfg["tar_file"]
+            run_args = d_cfg["run_args"]
```

Comment:
```suggestion
            run_args = d_cfg.get("run_args", "")
```
I get an error when this argument is not present. I think something like my suggestion should fix this, but haven't tested it